### PR TITLE
Update spatial4j, jts

### LINKED ISSUE #846: Update to Spatial4J 0.7 / JTS 1.15
Spatial4J 0.7 will contain JTS 1.15.

[See discussion](https://groups.google.com/d/msg/janusgraph-dev/JOQBE_0DNok/wkbDcl7JCgAJ)

- [x] Release date 12/22

Without much fanfare, [JTS](https://www.locationtech.org/projects/technology.jts) 1.15 was released.  This release uses the less restrictive EPL license which should enable JanusGraph to include it by default - IANAL.

- [Annoucement](https://dev.locationtech.org/mhonarc/lists/jts-dev/msg00215.html)
- [Release Notes](https://www.locationtech.org/projects/technology.jts/releases/1.15/plan)
- [License FAQ](https://github.com/locationtech/jts/blob/master/FAQ-LICENSING.md)
- [Upgrade Guide](https://github.com/locationtech/jts/blob/master/MIGRATION.md)
- [GitHub Release Page](https://github.com/locationtech/jts/releases/tag/jts-1.15.0)
- [Deployed to maven](http://search.maven.org/#artifactdetails%7Corg.locationtech.jts%7Cjts-core%7C1.15.0%7Cbundle)

Remove `optional` from pom(s).

### PR DESCRIPTION
**Pull request by robertdale**
Fixes #846

Signed-off-by: Robert Dale <robdale@gmail.com>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: janusgraph-core/src/main/java/org/janusgraph/core/attribute/JtsGeoshapeHelper.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ public JtsGeoshapeHelper() {
         binaryCodec = new JtsBinaryCodec(context, jtsFactory);
     }
 
-    public Geoshape geoshape(com.vividsolutions.jts.geom.Geometry geometry) {
+    public Geoshape geoshape(org.locationtech.jts.geom.Geometry geometry) {
```
Comment (sjudeng) [inline janusgraph-core/src/main/java/org/janusgraph/core/attribute/JtsGeoshapeHelper.java]: Add import for `org.locationtech.jts.geom.Geometry`

**Comment by amcp** [comment]
@robertdale please address @sjudeng comments, they do not seem to difficult to include in this PR.

**Comment by robertdale** [comment]
I had looked at it awhile back and what I seem to recall is that the storage format is different between the two. Simply merging them would render any non-jts data useless. Thus we would have to consider some migration path.  I'll try to take a look at it again to confirm.

**Comment by analytically** [comment]
Should we really consider a migration path when JanusGraph is still at 0.2?

**Comment by sjudeng** [comment]
So this looks good to me but since I contributed a commit we should probably get another approval before merge. Also @analytically the content here has changed a bit since your review in case you'd like to take another look.

**Comment by pluradj** [comment]
JTS is still required for Solr?

**Comment by sjudeng** [comment]
@pluradj With these updates the legacy vividsolutions JTS is not required on the client (JanusGraph) side but it might still be required on the server side depending on the Solr version being used. Also just a nit, JTS is still being used in core with these updates, it's just now the locationtech/Apache-licensed version and can therefore be included in the JanusGraph distribution by default. @robertdale Please correct me if I'm off on any of this.

**Comment by pluradj** [comment]
I mentioned Solr because I noticed that the ES and Lucene docs had updates, but not the [Solr](https://github.com/JanusGraph/janusgraph/blob/0.2/docs/solr.adoc) doc.

The "JTS ClassNotFoundException with Geo Data" section has a few references to the `com.vividsolutions` package. So if users are on an older version of Solr, it seems the requirement to pull down the JTS jars is still valid, at least for use with the Solr server. Solr 7.3 has the `org.locationtech` package via spatial4j 0.7, see [LUCENE-8161](https://issues.apache.org/jira/browse/LUCENE-8161) so going forward from that version should not have any requirement to download the JTS jars at all.

**Comment by sjudeng** [comment]
@pluradj I agree that note on "JTS ClassNotFoundException" that remains in the Solr docs is still valid, though maybe could be rephrased.

The earlier JTS note in the Geo section is no longer valid and is removed from the Solr docs in this PR though (see https://github.com/JanusGraph/janusgraph/pull/907/files#diff-9c7b18aa6600110edb0d594135518143).

**Pull request merged**